stm32f429i: ltdc display demo with text

brabo opened the pull request:
@H2OBrain nice work! I will test this out, if it works, and if it gets cleaned up a bit, we want to merge this into our fork ;) :+1:

h2obrain committed:
stm32f429i: ltdc display demo with text

h2obrain commented:
@brabo this demo needs a LOT of cleanup.. (and the balls are broken :)), also have a look at chucks graphics lib (https://github.com/ChuckM/simple-graphics)

twoerner committed:
cookie_io fixes for gcc-5.3

Using the latest (as of today) gcc-arm-embedded toolchain caused two build
failures similar to:

	error: unknown type name 'cookie_io_functions_t'

These custom IO functions are now protected by a define, so define _GNU_SOURCE
which enables all such areas. This is a libc issue.

Signed-off-by: Trevor Woerner <twoerner@gmail.com>

h2obrain committed:
Merge branch 'master' of https://github.com/libopencm3/libopencm3-examples into stm32_f4_new_ltdc_demo

h2obrain committed:
updated with remote..

h2obrain committed:
added bezier and linear functions

h2obrain committed:
initial version with flood_fill4 algorithm..

h2obrain committed:
small changes..

h2obrain committed:
added more history info about the creation of the gfx-library

h2obrain committed:
global scale mode to watch pixel perfectness..
+ fixes to draw_pixel and flood_fill4

h2obrain committed:
some optimisations, bug-fixes and comments

h2obrain committed:
balls!

h2obrain committed:
reorganised files and added matrix to do affine transformations..

h2obrain committed:
switched locm3 to new branch, fixed draw-rectangle

h2obrain committed:
fixed fly, since the rectangle fix, all rectangles are 1px less high

h2obrain committed:
changed libopencm3 submodule url..
